pallets/attestation/src/tests.rs

```diff
@@ -19,823 +19,980 @@
+	let mut attestation = generate_base_attestation::<Test>(attester.clone(), attester.clone());
 	attestation.delegation_id = Some(delegation_id);
 
 	let operation = generate_base_attestation_creation_details(claim_hash, attestation);
 
-	let ext = ctype_mock::ExtBuilder::default()
+	ExtBuilder::default()
+		.with_balances(vec![(attester.clone(), <Test as Config>::Deposit::get() * 100)])
 		.with_ctypes(vec![(operation.ctype_hash, attester.clone())])
-		.build(None);
-	let mut ext = delegation_mock::ExtBuilder::default()
 		.with_delegation_hierarchies(vec![(hierarchy_root_id, hierarchy_details, attester.clone())])
 		.with_delegations(vec![(delegation_id, delegation_node)])
-		.build(Some(ext));
-
-	ext.execute_with(|| {
-		assert_noop!(
-			Attestation::add(
-				get_origin(attester.clone()),
-				operation.claim_hash,
-				operation.ctype_hash,
-				operation.delegation_id
-			),
-			attestation::Error::<Test>::CTypeMismatch
-		);
-	});
+		.build()
+		.execute_with(|| {
+			assert_noop!(
+				Attestation::add(
+					Origin::signed(attester.clone()),
+					operation.claim_hash,
+					operation.ctype_hash,
+					operation.delegation_id
+				),
+				attestation::Error::<Test>::CTypeMismatch
+			);
+		});
 }
 
+// #############################################################################
 // submit_attestation_revocation_operation
 
 #[test]
 fn revoke_direct_successful() {
-	let revoker_keypair = get_alice_ed25519();
-	let revoker = get_ed25519_account(revoker_keypair.public());
+	let revoker: AccountIdOf<Test> = get_alice_ed25519().public().into();
 	let claim_hash = get_claim_hash(true);
-	let attestation = generate_base_attestation(revoker.clone());
+	let attestation = generate_base_attestation::<Test>(revoker.clone(), revoker.clone());
 
-	let operation = generate_base_attestation_revocation_details(claim_hash);
+	let operation = generate_base_attestation_revocation_details::<Test>(claim_hash);
 
-	let ext = ctype_mock::ExtBuilder::default()
+	ExtBuilder::default()
+		.with_balances(vec![(revoker.clone(), <Test as Config>::Deposit::get() * 100)])
 		.with_ctypes(vec![(attestation.ctype_hash, revoker.clone())])
-		.build(None);
-	let ext = delegation_mock::ExtBuilder::default().build(Some(ext));
-	let mut ext = ExtBuilder::default()
 		.with_attestations(vec![(operation.claim_hash, attestation)])
-		.build(Some(ext));
-
-	ext.execute_with(|| {
-		assert_ok!(Attestation::revoke(
-			get_origin(revoker.clone()),
-			operation.claim_hash,
-			operation.max_parent_checks
-		));
-	});
-
-	let stored_attestation =
-		ext.execute_with(|| Attestation::attestations(claim_hash).expect("Attestation should be present on chain."));
+		.build()
+		.execute_with(|| {
+			assert_ok!(Attestation::revoke(
+				Origin::signed(revoker.clone()),
+				operation.claim_hash,
+				operation.max_parent_checks
+			));
+			let stored_attestation =
+				Attestation::attestations(claim_hash).expect("Attestation should be present on chain.");
```

Comment:
W: Add check that the attestation can still be removed. I could not find this being checked, maybe I have overlooked it?

```suggestion
			assert_eq!(
				Balances::reserved_balance(revoker.clone()),
				<Test as Config>::Deposit::get()
			);
			assert_ok!(Attestation::revoke(
				Origin::signed(revoker.clone()),
				operation.claim_hash,
				operation.max_parent_checks
			));
			let stored_attestation =
				Attestation::attestations(claim_hash).expect("Attestation should be present on chain.");

			assert!(stored_attestation.revoked);
			assert_ok!(Attestation::remove(
				Origin::signed(revoker.clone()),
				claim_hash,
				operation.max_parent_checks
			));
			assert_eq!(Balances::reserved_balance(revoker.clone()), 0);
			assert!(Attestation::attestations(claim_hash).is_none());
```

Reply:
573ff483d2c3cf158d218f0ec0b11d82836c3ea3